### PR TITLE
use key into component registry instead of component class

### PR DESCRIPTION
**Pull request by davewo**
maintain get- and set- query api

### CODE REVIEW

**Comment by awkay** [comment]
Did you find the automated tests for this in the code???

**Comment by davewo** [comment]
yep I found them, I will keep this PR in draft as I rework the change based on our conversation

File: src/main/com/fulcrologic/fulcro/components.cljc [hunk 1]
```diff
@@ -799,13 +805,16 @@
...
-  (let [new-state (normalize-query-elements state-map query)
-        new-state (if (nil? (::queries new-state))
-                    (assoc new-state ::queries {})
-                    new-state)
-        top-query (link-query query)]
-    (if-let [queryid (some-> query meta :queryid)]
-      (util/deep-merge {::queries {queryid {:query top-query :id queryid}}} new-state)
+  (let [queryid       (some-> query meta :queryid)
+        component-key (class->registry-key (some-> query meta :component))
+        query'        (vary-meta query dissoc :queryid :component)
```
Comment (awkay) [inline src/main/com/fulcrologic/fulcro/components.cljc]: Why is this necessary? (clearing the metadata)...oh, is this so it does not end up in state map?

Comment (davewo) [reply src/main/com/fulcrologic/fulcro/components.cljc]: That was my thought, keeping metadata out of state. If you think its OK to leave it then I will update...
